### PR TITLE
cmd/utils: restore starting geth without any accounts and etherbase

### PR DESCRIPTION
**Pull request by fjl**

### CODE REVIEW

**Comment by robotally** [comment]
| Vote | Count | Reviewers |
| :---: | :---: | :---: |
| :+1: | 0 |  |
| :-1: | 0 |  |

_Updated: Tue Dec 1 10:22:14 UTC 2015_

File: cmd/utils/flags.go [hunk 1]
```diff
@@ -545,7 +545,7 @@ func MakeEtherbase(accman *accounts.Manager, ctx *cli.Context) common.Address {
 	}
 	// If no etherbase was specified and no accounts are known, bail out
 	accounts, _ := accman.Accounts()
-	if etherbase == "" && len(accounts) == 0 {
+	if (etherbase == "" || etherbase == "0") && len(accounts) == 0 {
```
Comment (karalabe) [inline cmd/utils/flags.go]: Shouldn't this rather simply be `etherbase == "" || len(accounts) == 0`?

I.e. if I haven't specified any, or there aren't any to be found?

Edit: Moved the comment into the PR... for some reason it went into a dangling commit previously.

**Comment by fjl** [comment]
PTAL, I changed the check to use `ctx.GlobalIsSet` instead.

**Comment by fjl** [comment]
```text
~/d/e/s/g/e/go-ethereum >> ./build/bin/geth --datadir=newdatadir
I1201 11:43:32.253464   31939 flags.go:543] WARNING: No etherbase set and no accounts found as default
...
~/d/e/s/g/e/go-ethereum >> ./build/bin/geth --datadir=newdatadir --etherbase=0
Fatal: Option "etherbase": can't get account #0 (open newdatadir/keystore: no such file or directory)
~/d/e/s/g/e/go-ethereum >> ./build/bin/geth --datadir=newdatadir --etherbase=foo
Fatal: Option "etherbase": invalid account address or index "foo"
```

**Comment by karalabe** [comment]
LGTM :+1:

**Comment by fjl** [comment]
Force-pushed one more time to restore running with `--etherbase=""`.

**Comment by obscuren** [comment]
:+1:

**Pull request merged**